components/scream/src/physics/shoc/atmosphere_macrophysics.cpp

```diff
@@ -162,15 +169,15 @@ void SHOCMacrophysics::initialize_impl (const util::TimeStamp& t0)
           qc_copy("qc_copy",m_num_cols,nlev_packs),
           tke_copy("tke_copy",m_num_cols,nlev_packs);
 
-  shoc_preprocess.set_variables(m_num_cols,m_num_levs,m_num_tracers,T_mid,
-                                z_int,z_mid,p_mid,pseudo_density,omega,phis,surf_sens_flux,surf_latent_flux,
-                                surf_u_mom_flux,surf_v_mom_flux,qv,qv_copy,qc,qc_copy,tke,tke_copy,
+  shoc_preprocess.set_variables(m_num_cols,m_num_levs,m_num_tracers,m_cell_area,
+                                T_mid,z_int,z_mid,p_mid,pseudo_density,omega,phis,surf_sens_flux,surf_latent_flux,
+                                surf_u_mom_flux,surf_v_mom_flux,qv,qv_copy,qc,qc_copy,tke,tke_copy,host_dx,host_dy,
                                 s,rrho,rrho_i,thv,dz,zt_grid,zi_grid,wpthlp_sfc,wprtp_sfc,upwp_sfc,vpwp_sfc,
                                 wtracer_sfc,wm_zt,exner,thlm,qw);
 
   // Input Variables:
-  input.host_dx     = m_shoc_fields_in["host_dx"].get_reshaped_view<const Real*>();
-  input.host_dy     = m_shoc_fields_in["host_dy"].get_reshaped_view<const Real*>();
+  input.host_dx     = shoc_preprocess.host_dx;
```

Comment:
Sorry for the late comment, but I was going through this once again, and stumbled into "host_dx". I am guessing 'host' stands for "host_atm_model", but in a GPU-friendly implementation, the word "host" can be a bit ambiguous. Would `column_dx` or `col_dx` or simply `dx` be an acceptable name? Pinging also @PeterCaldwell @bogensch .

Also, is it really needed to keep two views, one for dx and one for dy, considering we set dy=dx? Is there a plan to use a "non-square columns" assumption?

Reply:
Great points, Luca. In terms of dx and dy being combined into a single variable: I think I asked the same question and the answer is that this is a feature for the future. Folks might want to use SHOC in models with non-uniform x and y in the future.

In terms of "host" - I totally agree that this is confusing in a GPU code. I think `input_dx` would be more appropriate in this setting. @bogensch ?

Reply:
@tcclevenger what do you think of making dy a shallow copy of dx for now then? Granted, it's not _a ton_ of memory we'd save, but it's also a simple change. Keeping two aliasing views will make transitioning to dy!=dx easier later on (just need to change the way we set dy up, keeping the code that uses dy the same), while at the same time saving a bit of memory.

Reply:
Yeah, if host_dx is confusing then i'm fine with changing it.  I think I prefer "column_dx" over "input_dx", but both work.

Reply:
Ok, sounds good. I like the shallow copy idea and I don't really care what the name is. That said, since all variables are column variables (right?), I'm confused what value "column_" adds. Can we just call them dx and dy?

Reply:
No problem! I went ahead and changed `host_dx/dy` to `dx/dy` throughout shoc src code (I didn't change variable names related to testing, but can). 

For the shallow copy, instead of having both `dx/dy` as preprocessing variables, I just use `cell_length = sqrt(area)` with a comment that we would need to compute `dx/dy` there instead if `dx!=dy`.